bot/exts/utilities/challenges.py

```diff
@@ -0,0 +1,309 @@
+        """
+        Creates the 'other information embed' which displays miscellaneous information about the kata.
+
+        This embed shows statistics like the total number of people who completed the kata,
+        the total number of stars of the Kata, etc.
+
+        Takes in the kata information dictionary as an argument from the codewars.com API result (JSON)
+        """
+        miscellaneous_embed = Embed(
+            title="Other Information",
+            color=Colours.grass_green,
+        )
+        miscellaneous_embed.add_field(
+            name="`Total Score`",
+            value=f"**{kata_information['voteScore']}**",
+            inline=False,
+        )
+        miscellaneous_embed.add_field(
+            name="`Total Stars`",
+            value=f"**{kata_information['totalStars']}**",
+            inline=False,
+        )
+        miscellaneous_embed.add_field(
+            name="`Total Completed`",
+            value=f"**{kata_information['totalCompleted']}**",
+            inline=False,
+        )
+        miscellaneous_embed.add_field(
+            name="`Total Attempts`",
+            value=f"**{kata_information['totalAttempts']}**",
+            inline=False,
+        )
+        return miscellaneous_embed
+
+    @staticmethod
+    def create_view(dropdown: InformationDropdown, link: str) -> ui.View:
+        """
+        Creates the 'view', which is essentially a name for the discord UI.
+
+        The discord UI is implemented onto the embed, where the user can choose what information about the kata they
+        want, along with a link button to the kata itself.
+        """
+        view = ui.View()
+        # adding the dropdown
+        view.add_item(dropdown)
+        # manually creating the button as it is a link button and has no decorator
+        view.add_item(ui.Button(label="View the Kata", url=link))
+        return view
+
+    @commands.command(aliases=["kata"])
+    async def challenge(self, ctx: commands.Context, language: str = "python", *, query: str = None) -> None:
+        """
+        The challenge command pulls a random kata (challenge) from codewars.com.
+
+        The different ways to use this command are:
+        `.challenge <language>` - Pulls a random challenge within that language's scope.
+        `.challenge <language> <difficulty>` - The difficulty can be from 1-8,
+        1 being the hardest, 8 being the easiest. This pulls a random challenge within that difficulty & language.
+        `.challenge <language> <query>` - Pulls a random challenge with the query provided under the language
+        `.challenge <language> <query>, <difficulty>` - Pulls a random challenge with the query provided,
+        under that difficulty within the language's scope.
+        """
+        get_kata_link = f"https://codewars.com/kata/search/{language}"
+
+        if language and not query:
+            level = f"-{choice(['1', '2', '3', '4', '5', '6', '7', '8'])}"
```

Comment:
We can save some space by just having these as integers, when put into the f-string they will become formatted into strings.
```suggestion
            level = f"-{choice([1, 2, 3, 4, 5, 6, 7, 8])}"
```
That said, you can take a look at [`random.randint()`](https://docs.python.org/3/library/random.html#random.randint).